src/o365/spfx/commands/project/project-externalize.ts

```diff
@@ -0,0 +1,237 @@
+      }
+      else {
+        return null;
+      }
+    }
+  }
+
+  private getProjectVersion(): string | undefined {
+    const yoRcPath: string = path.resolve(this.projectRootPath as string, '.yo-rc.json');
+    if (fs.existsSync(yoRcPath)) {
+      try {
+        const yoRc: any = JSON.parse(fs.readFileSync(yoRcPath, 'utf-8'));
+        if (yoRc && yoRc['@microsoft/generator-sharepoint'] &&
+          yoRc['@microsoft/generator-sharepoint'].version) {
+          return yoRc['@microsoft/generator-sharepoint'].version;
+        }
+      }
+      catch { }
+    }
+
+    const packageJsonPath: string = path.resolve(this.projectRootPath as string, 'package.json');
+    try {
+      const packageJson: any = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+      if (packageJson &&
+        packageJson.dependencies &&
+        packageJson.dependencies['@microsoft/sp-core-library']) {
+        const coreLibVersion: string = packageJson.dependencies['@microsoft/sp-core-library'];
+        return coreLibVersion.replace(/[^0-9\.]/g, '');
+      }
+    }
+    catch { }
+
+    return undefined;
+  }
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-f, --outputFile [outputFile]',
+        description: 'Path to the file where the upgrade report should be stored in'
+      } as CommandOption
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    parentOptions.forEach(o => {
+      if (o.option.indexOf('--output') > -1) {
+        o.description = 'Output type. json|text|md. Default text';
+        o.autocomplete = ['json', 'text', 'md'];
+      }
+    });
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+      if (args.options.outputFile) {
+        const dirPath: string = path.dirname(path.resolve(args.options.outputFile));
+        if (!fs.existsSync(dirPath)) {
+          return `Directory ${dirPath} doesn't exist. Please check the path and try again.`;
+        }
+      }
+
+      return true;
+    };
+  }
+
+  public commandHelp(args: any, log: (help: string) => void): void {
+    const chalk = vorpal.chalk;
+    log(vorpal.find(commands.PROJECT_EXTERNALIZE).helpInformation());
+    log(
+      `   ${chalk.yellow('Important:')} Run this command in the folder where the project
+    that you want to upgrade is located. This command doesn't change your
+    project files.
+      
+  Remarks:
+
+    The ${this.name} command helps you externalize SharePoint Framework
+    project dependencies.  
+
+    This command doesn't change your project files. Instead, it gives you
+    a report with all steps necessary to externalize dependencies for your project.  
+
+  Examples:
+  
+    Get instructions to externalize dependencies for the current SharePoint 
+    Framework project and save the findings in a Markdown file
+      ${this.name} --output md --outputFile upgrade-report.md
```

Comment:
The .md file does have only `[object Object]` inside. This is screen from my notepad opened the report.
![image](https://user-images.githubusercontent.com/1049476/65518716-a4c42480-dedc-11e9-9fc7-286822cb604c.png)

Reply:
yes, I kind of butchered the md export when building this first draft. It's listed as a todo in the original comment on the PR. I'll leave this comment open for now until I fix it

Reply:
just fixed the report formats, closing the conversation, thanks!